src/Symfony/Component/Scheduler/Scheduler.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Scheduler;
+
+use Symfony\Component\Clock\Clock;
+use Symfony\Component\Clock\ClockInterface;
+use Symfony\Component\Scheduler\Generator\MessageGenerator;
+
+/**
+ * @experimental
+ */
+final class Scheduler
+{
+    /**
+     * @var array<MessageGenerator>
+     */
+    private array $generators = [];
+    private int $index = 0;
+    private bool $shouldStop = false;
+
+    /**
+     * @param iterable<Schedule> $schedules
+     */
+    public function __construct(
+        private array $handlers,
+        array $schedules,
+        private ClockInterface $clock = new Clock(),
+    ) {
+        foreach ($schedules as $schedule) {
+            $this->addSchedule($schedule);
+        }
+    }
+
+    public function addSchedule(Schedule $schedule)
+    {
+        $this->addMessageGenerator(new MessageGenerator($schedule, 'schedule_'.$this->index++, $this->clock));
+    }
+
+    public function addMessageGenerator(MessageGenerator $generator)
+    {
+        $this->generators[] = $generator;
+    }
+
+    /**
+     * Schedules messages.
+     *
+     * Valid options are:
+     *  * sleep (default: 1000000): Time in microseconds to sleep after no messages are found
```

Comment:
do we need the time in µs? the computation below only support seconds-level accuracy

Reply:
The answer is yes (to be consistent with Messenger Worker class)